src/envoy/mixer/grpc_transport.cc

```diff
@@ -113,7 +119,7 @@ GrpcTransport<RequestType, ResponseType>::GetFunc(Upstream::ClusterManager& cm,
             new Grpc::AsyncClientImpl<RequestType, ResponseType>(
                 cm, kMixerServerClusterName)),
         request, headers, response, on_done);
-    return [transport]() { transport->request_->cancel(); };
+    return [transport]() { transport->Cancel(); };
```

Comment:
This seems fine, but any reason to not use boxed pointers (shared/unique etc.) so that we don't have to deal with these?

Reply:
`transport` have longer lifetime than the lambda, otherwise proxy will simply crash.